types/types.cc

```diff
@@ -845,6 +846,52 @@ bool abstract_type::references_user_type(const sstring& keyspace, const bytes& n
     return find(*this, visitor{keyspace, name});
 }
 
+static std::unordered_set<sstring> get_all_referenced_user_types_aux(const abstract_type& t, bool unwrap_udt);
```

Comment:
I don't know if @piodul's suggestions will help or not, but I have to say that this code with all the helpers and auxiliaries, are completely opaque to me. Maybe some comments would have helped.